firebase.go

```diff
@@ -99,6 +99,11 @@ func New(url string) *Firebase {
 		CheckRedirect: redirectPreserveHeaders,
 	}
 
+	return NewFromHttpClient(url, client)
+}
+
+// NewFromHttpClient creates a new Firebase reference with given http client
+func NewFromHttpClient(url string, client *http.Client) *Firebase {
```

Comment:
I tagged a commit with the current constructor so people could use the old version. I would be ok with this breaking change and later updating the readme to explain the change.